src/command/build.rs

```diff
@@ -99,6 +101,143 @@ pub enum BuildProfile {
+            Self::Test(t) => format!("[[test]]\nname = {}", t),
+            Self::Benchmark(b) => format!("[[bench]]\nname = {}", b),
+        }
+    }
+}
+
+impl TryFrom<CargoTargetOptions> for CargoTarget {
+    type Error = Error;
+
+    fn try_from(opts: CargoTargetOptions) -> Result<Self, Self::Error> {
+        // the verbosity here (including all the None fields) is load bearing. it would be an
+        // invalid state to specify multiple targets, and we should let the user know that rather
+        // than failing silently.
+        match opts {
+            CargoTargetOptions {
+                binary: None,
+                example: None,
+                test: None,
+                bench: None,
+            } => Ok(Self::Library),
+            CargoTargetOptions {
+                binary: Some(bin),
+                example: None,
+                test: None,
+                bench: None,
+            } => Ok(Self::Binary(bin)),
+            CargoTargetOptions {
+                binary: None,
+                example: Some(example),
+                test: None,
+                bench: None,
+            } => Ok(Self::Example(example)),
+            CargoTargetOptions {
+                binary: None,
+                example: None,
+                test: Some(test),
+                bench: None,
+            } => Ok(Self::Test(test)),
+            CargoTargetOptions {
+                binary: None,
+                example: None,
+                test: None,
+                bench: Some(bench),
+            } => Ok(Self::Benchmark(bench)),
+            CargoTargetOptions {
+                binary,
+                example,
+                test,
+                bench,
+            } => {
+                let mut requested = Vec::new();
+
+                if let Some(b) = binary {
+                    requested.push(Self::Binary(b));
+                }
+
+                if let Some(e) = example {
+                    requested.push(Self::Example(e));
+                }
+
+                if let Some(t) = test {
+                    requested.push(Self::Test(t));
+                }
+
+                if let Some(b) = bench {
+                    requested.push(Self::Benchmark(b));
+                }
+
+                bail!("Can only specify one Cargo target. Attempted to select multiple targets at the same time: {:?}", requested)
+            }
+        }
+    }
+}
+
+/// Specific, mutually exclusive CLI options for Cargo target selection
+#[derive(Debug, Default, StructOpt)]
+pub struct CargoTargetOptions {
+    #[structopt(long = "bin", group = "cargo-target")]
+    /// Build only the specified binary.
+    pub binary: Option<String>,
+
+    #[structopt(long = "example", group = "cargo-target")]
+    /// Build only the specified example.
+    pub example: Option<String>,
+
+    #[structopt(long = "test", group = "cargo-target")]
+    /// Build only the specified test target.
+    pub test: Option<String>,
```

Comment:
Not quite. (If this worked, see the last bullet in the PR description) it would build and package the specified test in the `./tests` directory as a WASM binary with JS bindings, rather than executing it.